src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/3.sql

```diff
@@ -1,33 +1,24 @@
--- NOTE: This script DROPS AND RECREATES all database objects.
 -- Style guide: please see: https://github.com/ktaranov/sqlserver-kit/blob/master/SQL%20Server%20Name%20Convention%20and%20T-SQL%20Programming%20Style.md
-
 /*************************************************************
-    Configure database
+Wrapping up in the multiple transactions except CREATE FULLTEXT INDEX which is non-transactional script.
+Guidelines to create scripts - https://github.com/microsoft/healthcare-shared-components/tree/master/src/Microsoft.Health.SqlServer/SqlSchemaScriptsGuidelines.md
 **************************************************************/
 
--- Enable RCSI
-IF ((SELECT is_read_committed_snapshot_on FROM sys.databases WHERE database_id = DB_ID()) = 0) BEGIN
-    ALTER DATABASE CURRENT SET READ_COMMITTED_SNAPSHOT ON
-END
+SET XACT_ABORT ON
 
--- Avoid blocking queries when statistics need to be rebuilt
-IF ((SELECT is_auto_update_stats_async_on FROM sys.databases WHERE database_id = DB_ID()) = 0) BEGIN
-    ALTER DATABASE CURRENT SET AUTO_UPDATE_STATISTICS_ASYNC ON
-END
-
--- Use ANSI behavior for null values
-IF ((SELECT is_ansi_nulls_on FROM sys.databases WHERE database_id = DB_ID()) = 0) BEGIN
-    ALTER DATABASE CURRENT SET ANSI_NULLS ON
+BEGIN TRANSACTION
+/***********************************************************************
+ NOTE: just checking first object, since this is run in transaction 
```

Comment:
This is better and simpler than making every script idempotent in a transaction. Thanks for doing that.